### PR TITLE
52 create help button

### PR DESCRIPTION
**Pull request by will-weiss**
Addresses #52, which has been broken up further into #72 and #73. This still doesn't handle the styling of the button and doesn't take a screenshot, but #73 is a larger task that will require a fair bit of work, so I figured we should try to get this merged into `main`.

ezgif-7-ffcb3cbb5255

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -33,54 +33,85 @@ type ScreenshotState = {
...
-  handle: string | null
-}
-
 type EditingScreenshotState = {
   color: string
   screenshot: Screenshot
 }
 
 type FeedbackState = {
+  isTweeting: boolean
```
Comment (shachar-langer) [inline src/types.ts:42]: Why did you move `isTweeting` inside the feedback?

File: src/background/responders.ts [hunk 2]
```diff
@@ -131,10 +109,11 @@ export const responders: Responders<Action> = {
     nextTabs.set(tab.id, {
       ...tab,
       feedbackState: {
```
Comment (shachar-langer) [inline src/background/responders.ts]: I think we can call `updateActiveFeedback` every time we update the feedback state

File: src/popup/app.tsx [hunk 3]
```diff
@@ -1,42 +1,29 @@
...
+    case 'Authenticating':
+      return <views.Authenticating {...appState} />
+    case 'Authenticated':
+      return <views.Authenticated {...appState} />
   }
 }
+
+export function App(props: AppProps): JSX.Element {
+  return (
+    <div className="app">
```
Comment (shachar-langer) [inline src/popup/app.tsx:25]: Wrapping all views with this div cause CSS issues. For example, when adding more than 3 screenshots.

File: src/popup/views/authenticated.tsx [hunk 4]
```diff
@@ -6,33 +6,24 @@ import { EmojiPicker } from '../components/emoji-picker'
...
+}: AuthenticatedState): JSX.Element | null {
+  if (tweeting) {
+    return <Tweeting at={tweeting.at} />
   }
 
-  if (feedback.editingScreenshot) {
-    return <EditingScreenshot color={feedback.editingScreenshot.color} screenshot={feedback.editingScreenshot.screenshot} />
+  if (!feedback.exists) {
+    if (feedback.reasonDisabledMessage) {
+      return <p>{feedback.reasonDisabledMessage}</p>
```
Comment (shachar-langer) [inline src/popup/views/authenticated.tsx:24]: We should add a ticket to style this message.

File: src/tests/happy-path.test.ts [hunk 5]
```diff
@@ -33,14 +33,20 @@ describe('happy path', () => {
...
+      expect(state.focusedWindowId).to.equal(-1)
+      expect(state.tabs).to.be.an.instanceOf(Map)
+      expect(state.tabs).to.have.property('size', 0)
+      expect(state.auth).to.eql({ state: 'not_authed' })
+      expect(state.pickingEmoji).to.equal(false)
+      expect(state.help.on).to.equal(false)
+      expect(state.help.feedbackState).to.be.an('object')
+      expect(state.help.feedbackState.editingScreenshot).to.equal(null)
+      expect(state.help.feedbackState.screenshots).to.eql([])
+      expect(getPlainText(state.help.feedbackState.editorState)).to.equal('@roarmhi ')
```
Comment (shachar-langer) [inline src/tests/happy-path.test.ts:46]: Why does it load the "help" section? It should load an empty tab state, no?

**Review by shachar-langer — COMMENTED** [review]
The last commit looks good

File: src/types.ts (hunk 1 above)
Comment (will-weiss) [reply src/types.ts:43]: So that this was a one-liner

```
return updateActiveFeedback(state, () => ({ isTweeting: true }))
```

File: src/background/responders.ts [hunk 6]
```diff
@@ -283,10 +217,10 @@ export const responders: Responders<Action> = {
...
   },
   'chrome.windows.onRemoved'(state, { windowId }): Partial<StoreState> {
-    const tabs = new Map(state.tabs)
+    let tabs = state.tabs // tslint:disable-line:no-let
     for (const [id, tab] of state.tabs.entries()) {
       if (tab.windowId === windowId) {
-        tabs.delete(id)
+        tabs = tabs.delete(id)
       }
     }
```
Comment (shachar-langer) [inline src/background/responders.ts:225]: 👌

**Review by shachar-langer — APPROVED** [review]
